torch/testing/_internal/common_methods_invocations.py

```diff
@@ -1337,6 +1359,27 @@ def reference_sigmoid(x):
                                     dtypes=[torch.bfloat16]),
                        )
                        ),
+        UnaryUfuncInfo('lgamma',
+                       ref=reference_lgamma,
+                       decorators=(precisionOverride({torch.float16: 7e-1}),),
+                       dtypes=all_types_and(torch.bool),
+                       dtypesIfCPU=all_types_and(torch.bool, torch.bfloat16),
+                       dtypesIfCUDA=all_types_and(torch.bool, torch.half),
+                       skips=(
+                           # Reference: https://github.com/pytorch/pytorch/pull/50140#discussion_r552615345
+                           SkipInfo('TestUnaryUfuncs', 'test_reference_numerics',
+                                    dtypes=[torch.bfloat16]),
+                           # Reference: https://github.com/pytorch/pytorch/pull/50140#issuecomment-756150214
+                           SkipInfo('TestUnaryUfuncs', 'test_reference_numerics',
+                                    dtypes=[torch.float32, torch.float64], active_if=IS_WINDOWS),
+                           # Backward of `lgamma` uses `digamma` but `digamma`
```

Comment:
This is interesting. From the perspective of this PR this skip is correct, but I thought test_variant_consistency_jit wrapped the backward call in a try/except block to catch these errors?

Reply:
I have verified. I don't see any try/execpt block in the failing test.

<details>

<summary>Stack Trace </summary>

```
======================================================================
ERROR: test_variant_consistency_jit_lgamma_cpu_bfloat16 (__main__.TestCommonCPU)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/kshiteej/.conda/envs/PytorchENV/lib/python3.7/site-packages/torch/testing/_internal/common_device_type.py", line 284, in instantiated_test
    raise rte
  File "/home/kshiteej/.conda/envs/PytorchENV/lib/python3.7/site-packages/torch/testing/_internal/common_device_type.py", line 279, in instantiated_test
    result = test_fn(self, *args)
  File "/home/kshiteej/.conda/envs/PytorchENV/lib/python3.7/site-packages/torch/testing/_internal/common_device_type.py", line 253, in test_wrapper
    return test(*args, **kwargs)
  File "test/test_ops.py", line 290, in test_variant_consistency_jit
    no_grad=(dtype not in dtypes_to_grad_check))
  File "/home/kshiteej/.conda/envs/PytorchENV/lib/python3.7/site-packages/torch/testing/_internal/common_jit.py", line 77, in check_against_reference
    allow_unused=allow_unused)
  File "/home/kshiteej/.conda/envs/PytorchENV/lib/python3.7/site-packages/torch/autograd/__init__.py", line 225, in grad
    inputs, allow_unused, accumulate_grad=False)
RuntimeError: "digamma" not implemented for 'BFloat16'

----------------------------------------------------------------------
Ran 94 tests in 5.282s

FAILED (errors=1, skipped=8)
```

</details>

Code fails in `check_against_reference`
https://github.com/pytorch/pytorch/blob/5546a12fe31eb3b2254eeb9837e4f1e6b000f256/test/test_ops.py#L272-L300

The code for `check_against_reference` also doesn't have any try/except
https://github.com/pytorch/pytorch/blob/5546a12fe31eb3b2254eeb9837e4f1e6b000f256/torch/testing/_internal/common_jit.py#L39-L108


**Note**:
The `test_variant_consistency_eager` test has a try/except block which checks if the eager mode raises or not.
https://github.com/pytorch/pytorch/blob/5546a12fe31eb3b2254eeb9837e4f1e6b000f256/test/test_ops.py#L183-L237

Reply:
Aha! That's why I was confused. Thank you for verifying. I'll update the test suite tracker.